### ISSUE TITLE
Help - force www or "non www" version of a site 

### ISSUE THREAD

**Issue by andreliem** [nuxt/nuxt.js#1970]
Does anyone have any tips or snippets which can force a specific domain? Either forcing the www or non www version? Is there anything like the `redirect-ssl` that could be used? 

Or would you just recommend handling this at the DNS level?

**Comment by KonstantinVlasov** [comment]
I think you should do it at the DNS or server level (i.e. nginx)

**Comment by andreliem** [comment]
Thanks, yeah that's my conclusion ;)

**Comment by andreliem** [comment]
So actually I've changed my mind. Doing this at the DNS level is not great for me, the DNS I'm using could only redirect at the root level but not with paths. 

e.g. - 
`domain.com redirects to www.domain.com = OK`
`domain.com/?query=param to www.domain.com/?query=param = FAIL`

So I had to do this at the app level. I'm running on Heroku so it's not like I have easy access to the server config.

In short, I've relied on `serverMiddleware`. I'll probably post up a small repo for this but the snippet for those interested is something like this.

```
module.exports = function (req, res, next) {
  const host = req.headers.host
  const url = req.url
  const env = process.env.NODE_ENV
  const forceDomain = 'https://www.DOMAIN.com'

  if (env === 'production' && host !== 'www.DOMAIN.com') {
    res.writeHead(301, { Location: forceDomain + url })
    return res.end()
  }

  return next()
}
```

Then you just need to register this in `serverMiddleware` in the `nuxt.config.js` file.
